LanguageFeatures/Patterns/logical_or_A04_t02.dart

```diff
@@ -27,33 +27,46 @@ import "patterns_lib.dart";
 import "../../Utils/expect.dart";
 
 main() {
-  Shape shape1 = Square.withLog(1);
+  Shape shape1 = Square(1);
   switch (shape1) {
-    case Circle(area: var s) | Square(area: var s):
-      Expect.equals("Square.area", shapesLog);
+    case Square(area: 2) | Square(area: 1):
+      Expect.equals("Square.area=2;Square.area=1;", shape1.log);
       break;
     default:
       print("Other");
   }
 
-  Shape shape2 = Shape.withLog();
-  clearLog();
+  Shape shape2 = Shape();
   switch (shape2) {
-    case Square(area: var s) | Rectangle(area: var s) | Shape(area: var s):
-      Expect.equals("Shape.area", shapesLog);
+    case Square(area: 2) | Rectangle(area: 1) | Shape(area: 0):
+      Expect.equals("Shape.area=2;Shape.area=1;Shape.area=0;", shape2.log);
```

Comment:
I can't immediately see how this would produce a log as stated. Wouldn't it simply write 'Shape.area=0' into the log after having failed already at the type test of the first two patterns (which wouldn't log anything)?

Reply:
No, it's the other way around: We're considering an object `o` whose run-time type is `Shape`, and we're testing whether `o` has type `Square`. The answer is "no", because we do not have `Shape <: Square`, we only have `Square <: Shape`.